metrics-micrometer/src/main/java/org/axonframework/micrometer/PayloadTypeMessageMonitorWrapper.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2010-2018. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.micrometer;
+
+import org.axonframework.messaging.Message;
+import org.axonframework.monitoring.MessageMonitor;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+/**
+ * A {@link MessageMonitor} implementation which creates a new MessageMonitor for every {@link Message} payload type
+ * ingested by it. The PayloadTypeMessageMonitorWrapper keeps track of all distinct payload types it has created
+ * and only creates a new one if there is none present.
+ * <p>
+ * The type of MessageMonitor which is created for every payload type is configurable, as long as it implements
+ * MessageMonitor}.
+ *
+ * @param <T> The type of the MessageMonitor created for every payload type.Must implement both {@link MessageMonitor}
+ *
+ * @author Marijn van Zelst
```

Comment:
I just now see I never added an author and since tag to the original when I implemented this...thanks for the reminder.

Reply:
Ah, I've added you as an author :smile: